oauth2.py

```diff
@@ -51,7 +51,12 @@ def refresh_token(self, refresh_token: str) -> json:
                 'refresh_token': refresh_token}
         response = requests.post(self.oauth_token_url, data=body, headers=self.encode_header())
         token = json.loads(response.content.decode('utf-8'))
-        self.save_token(token, refresh_token=refresh_token)
+        try:
+            assert token['refresh_token'] is not None
+            self.save_token(token)
+        except (KeyError, AssertionError):
```

Comment:
Is `(KeyError, AssertionError)` sufficiently wide?

Reply:
Yes.